src/torque-offset-estimator.cpp

```diff
@@ -126,84 +125,83 @@ namespace dynamicgraph
         else { //sensor_offset_status == COMPUTED
           SEND_MSG("Recomputing offset with no. iterations:"+ nIterations, MSG_TYPE_DEBUG);
 
-          stdVecJointTorqueOffsets.clear();
+          // stdVecJointTorqueOffsets.clear();
+          current_progress = 0;
           jointTorqueOffsets.setZero();
 
           n_iterations = nIterations;
-          epsilon = _epsilon;
+          epsilon = epsilon_;
           sensor_offset_status = INPROGRESS;
         }
         return;
       }
       
       DEFINE_SIGNAL_INNER_FUNCTION(collectSensorData, dummy)
       {
-        if (sensor_offset_status == PRECOMPUTATION)  {
-          SEND_MSG("Currently Precomputation. Not collecting signals", MSG_TYPE_DEBUG);
-        }
-        else if (sensor_offset_status == INPROGRESS) {
+        if (sensor_offset_status == INPROGRESS) {
           // Check the current iteration status
-          int _i = stdVecJointTorqueOffsets.size();
+          int i = current_progress;
           
-          if (_i < n_iterations) {
-            SEND_MSG("Collecting signals for iteration no:" + _i , MSG_TYPE_DEBUG_STREAM);
+          if (i < n_iterations) {
+            SEND_MSG("Collecting signals for iteration no:" + i , MSG_TYPE_DEBUG_STREAM);
 
             const Eigen::VectorXd& sot_enc = m_base6d_encodersSIN(iter);
             const Eigen::VectorXd& IMU_acc = m_accelerometerSIN(iter);
-            const Eigen::VectorXd& IMU_gyr = m_gyroscopeSIN(iter);
-            const Eigen::VectorXd& _tau = m_jointTorquesSIN(iter);
+            const Eigen::VectorXd& tau = m_jointTorquesSIN(iter);
 
-            Eigen::VectorXd _enc(m_model.nq);
-            base_sot_to_urdf(sot_enc.head<6>(), _enc.head<7>());
-            _enc.tail(m_model.njoints) = sot_enc.tail(m_model.njoints);
+            Eigen::VectorXd enc(m_model.nq);
+            enc.tail(m_model.nv-6) = sot_enc.tail(m_model.nv-6);
```

Comment:
I think here you should use the function joint_sot_to_urdf to convert the joint order from sot to pinocchio convention